routers/user/setting/profile.go

```diff
@@ -141,11 +141,14 @@ func UpdateAvatarSetting(ctx *context.Context, form auth.AvatarForm, ctxUser *mo
 		if err = ctxUser.UploadAvatar(data); err != nil {
 			return fmt.Errorf("UploadAvatar: %v", err)
 		}
-	} else if ctxUser.UseCustomAvatar && !com.IsFile(ctxUser.CustomAvatarPath()) {
-		// No avatar is uploaded but setting has been changed to enable,
-		// generate a random one when needed.
-		if err := ctxUser.GenerateRandomAvatar(); err != nil {
-			log.Error("GenerateRandomAvatar[%d]: %v", ctxUser.ID, err)
+	} else if ctxUser.UseCustomAvatar {
+		fs := storage.Avatar(ctx.Req.Context(), ctxUser.Avatar)
```

Comment:
Why not use `CustomAvatarPath` any more?

Reply:
@lunny we don't use it anymore https://github.com/go-gitea/gitea/pull/9567#discussion_r362365278, the path is handled by storage.OpenBucket